packages/alchemy/src/client/types.ts

```diff
@@ -49,43 +72,68 @@ export type AlchemyRpcSchema = [
   {
     Method: "rundler_maxPriorityFeePerGas";
     Parameters: [];
-    ReturnType: Hex;
+    ReturnType: UserOperationRequest<EntryPointVersion>["maxPriorityFeePerGas"];
   }
 ];
 
 export type ClientWithAlchemyMethods = BundlerClient<HttpTransport> & {
-  request: BundlerClient["request"] &
+  request: BundlerClient<HttpTransport>["request"] &
     {
       request(args: {
         method: "alchemy_requestPaymasterAndData";
         params: [
           {
             policyId: string;
             entryPoint: Address;
-            userOperation: UserOperationRequest;
+            userOperation: UserOperationRequest<EntryPointVersion>;
           }
         ];
-      }): Promise<{ paymasterAndData: Hex }>;
+      }): Promise<
+        | {
+            paymasterAndData: UserOperationRequest<"0.6.0">["paymasterAndData"];
+          }
+        | Pick<
+            UserOperationRequest<"0.7.0">,
+            | "paymaster"
+            | "paymasterData"
+            | "paymasterPostOpGasLimit"
+            | "paymasterVerificationGasLimit"
+          >
+      >;
 
       request(args: {
         method: "alchemy_requestGasAndPaymasterAndData";
         params: [
           {
             policyId: string;
             entryPoint: Address;
-            userOperation: UserOperationRequest;
+            userOperation: UserOperationRequest<EntryPointVersion>;
             dummySignature: Hex;
-            overrides?: RequestGasAndPaymasterAndDataOverrides;
+            overrides?: RequestGasAndPaymasterAndDataOverrides<EntryPointVersion>;
           }
         ];
-      }): Promise<{
-        paymasterAndData: Hex;
-        callGasLimit: Hex;
-        verificationGasLimit: Hex;
-        preVerificationGas: Hex;
-        maxFeePerGas: Hex;
-        maxPriorityFeePerGas: Hex;
-      }>;
+      }): Promise<
+        Pick<
+          UserOperationRequest<EntryPointVersion>,
+          | "callGasLimit"
+          | "preVerificationGas"
+          | "verificationGasLimit"
+          | "maxFeePerGas"
+          | "maxPriorityFeePerGas"
+        > &
+          (
+            | {
+                paymasterAndData: UserOperationRequest<"0.6.0">["paymasterAndData"];
+              }
+            | Pick<
+                UserOperationRequest<"0.7.0">,
+                | "paymaster"
+                | "paymasterData"
+                | "paymasterPostOpGasLimit"
+                | "paymasterVerificationGasLimit"
+              >
+          )
```

Comment:
nit: instead of copy pasting this type, can we wrap this in a different type definition so it can be reused?